src/gtc/gtcpp/gtcpp_codegen.py

```diff
@@ -84,8 +98,21 @@ def visit_AccessorRef(self, accessor_ref: gtcpp.AccessorRef, **kwargs):
             k_offset = self.visit(accessor_ref.offset.k, **kwargs)
         else:
             raise TypeError("Unsupported offset type")
-        data_index = "".join(f", {self.visit(d)}" for d in accessor_ref.data_index)
-        return f"eval({accessor_ref.name}({i_offset}, {j_offset}, {k_offset}{data_index}))"
+
+        if accessor_ref.name in temp_decls and accessor_ref.data_index:
+            temp = temp_decls[accessor_ref.name]
```

Comment:
Can you retrieve the value from the symbol table similar to what you are doing in the CUIR codegen?